src/process_controller.cpp

```diff
@@ -189,84 +191,77 @@ void ProcessController::handle_ptrace_event(pid_t child, int event)
         try
         {
             // we need the tid of the new process
-            pid_t new_tid = ptrace_geteventmsg(child);
+            Thread new_thread = Thread(ptrace_geteventmsg(child));
 
             // Thread may have been clone from another thread, figure out which
             // process they both belong too.
-            pid_t pid = groups_.get_group(ExecutionScope::thread(child)).tid();
-            std::string command = get_task_comm(pid, new_tid);
-            Log::info() << "New thread " << new_tid << " (" << command << "): cloned from " << child
-                        << " in process " << pid;
+            Process process = groups_.get_process(new_thread);
+            std::string command = get_task_comm(process, new_thread);
+            Log::info() << "New " << new_thread.name() << " (" << command << "): cloned from "
+                        << child.name() << " in " << process.name();
 
             // Register the newly created thread for monitoring:
             // (1) Keep track of the process the new thread was spawned in.
-            groups_.add_child(ExecutionScope::thread(new_tid), ExecutionScope::thread(pid));
+            groups_.add_thread(new_thread, process);
```

Comment:
This line is odd. If you look how `process` is defined, it's basically:

`groups_.add_thread(new_thread, groups_.get_process(new_thread));`

Either, I'm totally wrong, `process` is undefined, or this line is redundant.